### ISSUE TITLE
Cache NumberTextStyle character attributes?

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/DiffableTextViews#103]
Attributes are constant and can be cached.
Pros: no if/else and fewer lookups per snapshot
Cons: at least 238 bytes per cached format scheme

**Comment by oscbyspro** [comment]
I think the same can be said for the translation, at which point there should be no need to store components separately and it would be better to store only a translation map and an attributes map. I haven’t thought about it fully, yet. It’s just something I noticed.

**Comment by oscbyspro** [comment]
Proposed solution is available in a847994c146e72fee96b6ffa508a4aa1d3b313b8
---

Reader was reworked as a container of:

```
Components: Sign, Digit, Separator <-> Character
Attributes: Character -> Attribute
Translator: Character -> Character === ASCII, localized -> localized
```

where Components is Lexicon renamed.
